src/java/org/apache/cassandra/repair/state/StreamState.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.cassandra.repair.state;
+
+import java.net.InetSocketAddress;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.apache.cassandra.streaming.StreamEvent;
+import org.apache.cassandra.streaming.StreamEventHandler;
+import org.apache.cassandra.streaming.StreamManager;
+import org.apache.cassandra.streaming.StreamResultFuture;
+
+//TODO make thread safe
+public class StreamState extends AbstractState<StreamState.State> implements StreamEventHandler
+{
+    public enum State { START }
+
+    public final Phase phase = new Phase();
+    private final boolean follower;
+    private Set<InetSocketAddress> peers;
+
+    public StreamState(StreamResultFuture future)
+    {
+        super(future.planId, State.class);
+        this.follower = future.getCoordinator().isFollower();
+        this.peers = future.getCoordinator().getPeers();
+    }
+
+    public boolean isFollower()
+    {
+        return follower;
+    }
+
+    public Set<InetSocketAddress> getPeers()
+    {
+        return peers;
+    }
+
+    //TODO rewrite
+    @Override
+    public float getProgress()
+    {
+        return super.getProgress();
+    }
+
+    @Override
+    public void handleStreamEvent(StreamEvent event)
+    {
+        StreamResultFuture stream = StreamManager.instance.getReceivingStream(event.planId);
```

Comment:
should we check `initiatorStream` as well?

Reply:
this was fixed in the streaming table patch.